src/tests/BuildWasmApps/Wasm.Build.Tests/WasmRunOutOfAppBundleTests.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class WasmRunOutOfAppBundleTests : BuildTestBase
+    {
+        public WasmRunOutOfAppBundleTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext) : base(output, buildContext)
+        {}
+
+        [Theory]
+        [BuildAndRun]
+        public void RunOutOfAppBundle(BuildArgs buildArgs, RunHost host, string id)
+        {
+            buildArgs = buildArgs with { ProjectName = $"outofappbundle_{buildArgs.Config}_{buildArgs.AOT}" };
+            buildArgs = ExpandBuildArgs(buildArgs);
+
+            BuildProject(buildArgs,
+                            id: id,
+                            new BuildProjectOptions(
+                                InitProject: () => File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), s_mainReturns42),
+                                DotnetWasmFromRuntimePack: !(buildArgs.AOT || buildArgs.Config == "Release")));
+
+            string binDir = GetBinDir(baseDir: _projectDir!, config: buildArgs.Config);
+            string appBundleDir = Path.Combine(binDir, "AppBundle");
+            string tmpBundleDirName = "AppBundleTmp";
+            string tmpBundleDir = Path.Combine(binDir, tmpBundleDirName);
```

Comment:
What do you think of creating this tmpBundleDir in a completely different location, like under `Path.GetTempPath()`? This would ensure that it isn't dependent on anything in the binDir.
And make sure to add `id`, and the project name to the dir name, so the different runs don't overlap.

Reply:
It is a good method in Node and V8 environments!
I' ll change that style.

However, browsers like Chrome cannot explore the parent directory, so I would like to keep the current style.